### ISSUE TITLE
v17.1.4 throws errors with TypeScript 2.4.1

### ISSUE THREAD

**Issue by eliashdezr** [DevExpress/devextreme-angular#502]
**To help us process the issue more efficiently, please provide the following information:**
- Specify the version of the *devextreme-angular* and *devextreme* packages you are using.    
    devexteme version: 17.1.4
    devextreme-angular version: 17.1.4

- Specify the type of the issue (check one with "x"): 
[x ] bug Report

Just adding the dependency to an angular project created with the Angular CLI tool and using TypeScript 2.4.1

```
ERROR in C:/X/node_modules/devextreme-angular/core/nested-option.d.ts (20,31): Class 'BaseNestedOption' incorrectly implements interface 'ICollectionNestedOptionContainer'.
  Types of property 'setChildren' are incompatible.
    Type '<T extends ICollectionNestedOption>(propertyName: string, items: QueryList<T>) => any' is not assignable to type '<T>(propertyName: string, items: QueryList<T>) => any'.
      Types of parameters 'items' and 'items' are incompatible.
        Type 'QueryList<T>' is not assignable to type 'QueryList<ICollectionNestedOption>'.
          Type 'T' is not assignable to type 'ICollectionNestedOption'.
```

**Comment by AndrewVetovitz** [comment]
Roll back to 2.3.2 for now, this is stable.

**Comment by webcat12345** [comment]
Mine is typescript 2.4.2 and `devextreme-angular` version is 17.1.4 throws same error.
Downgraded typescript version to 2.3.2 and it works.

Is there any scheduled update regarding to this error?

**Comment by GoshaFighten** [comment]
@webcat12345 
This issue was fixed in version 17.1.5-rc.1. See #506 .
